### ISSUE TITLE
Namespace created explicitly gets deleted even though CreateNamespace=true is set on an Application

### ISSUE THREAD

**Issue by dominykas** [argoproj/argo-cd#16995]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

(the problem described in #15441 has not been actually fixed - but I have no access to re-open the existing issue)

I'm migrating to use `managedNamespaceMetadata` in `Application` resources and I'd like to remove explicit `Namespace` resources from configuration and rely solely on `CreateNamespace=true`, however there does not seem to be a migration path.

I've duplicated all the annotations, so the `Namespace` and the `managedNamespaceMetadata` on the `Application` are the same and I'm now ready to remove the `Namespace`, however doing so actually deletes the namespace and recreates it during the Sync.

I would expect the application and the namespace to continue existing as is.

I'm fairly certain that this used to work at some point in the past, because I've already done a similar migration where I've replaced `Namespace` with `CreateNamespace=true`, but that was before `managedNamespaceMetadata` was a feature.

**To Reproduce**

1. Have an app of apps
2. For each app have a `Namespace` and an `Application` resource with `CreateNamespace=true`
3. Remove the `Namespace` from gitops

**Expected behavior**

The existing `Namespace` should stay intact, as it should now be handled by the `Application`.

**Version**

I'm not logged in via the CLI, but here's what I have in UI: `v2.9.5+f943664`

**Comment by nmohoric** [comment]
I believe the only way to do this is to remove the `app.kubernetes.io/instance` label from the namespace by hand via kubectl before removing the namespace from gitops.

**Comment by dominykas** [comment]
Yeah, and alternatively maybe it can be made to work by setting `Prune=false` on the current application - I might actually try that out.

Nonetheless, this is still very unexpected and dangerous.

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and let us know if the issue is still present, please?

**Comment by dominykas** [comment]
I had tested this on 2.11 (and possibly on 2.12 - can't recall) - but overall we no longer plan on using this, so it's unlikely we'll be revisiting it again.
